### PR TITLE
Rename WASM ABI functions to wasi-standard snake_case.

### PR DESCRIPTION
**Pull request by jplevyak**
This follows from the previous PRs.  See the last commit for the specific changes.

### CODE REVIEW

File: test/extensions/wasm/test_data/logging_rust/src/lib.rs [hunk 1]
```diff
@@ -10,14 +10,14 @@ extern "C" {
 
 #[no_mangle]
 #[allow(non_snake_case)]
```
Comment (PiotrSikora) [inline test/extensions/wasm/test_data/logging_rust/src/lib.rs]: You can remove `#[allow(non_snake_case)]` here and in other functions.

**Comment by PiotrSikora** [comment]
Nit: it's more of a Wasm than WASI naming convention.

File: test/extensions/wasm/test_data/logging_rust/src/lib.rs (hunk 1 above)
Comment (jplevyak) [reply test/extensions/wasm/test_data/logging_rust/src/lib.rs]: done

**Comment by PiotrSikora** [comment]
Can you merge master? Thanks!

**Pull request merged**